src/NextGen/mParticleCallback.ts

```diff
@@ -0,0 +1,16 @@
+export abstract class mParticleCallback<Success, Error> {
+    public isSuccessful: boolean = false;
+    public isCompleted: boolean = false;
+
+    // QUESTION: Why do these methods have a result param that isn't used?
```

Comment:
I agree - seems like there should be a `this.result = result` in order to allow access to what is on the result (assuming there is anything of value there).

Reply:
So as we discussed on the call, I strongly believe that we should use function/lambda callbacks on Web and iOS as it's much more idiomatic. The way this part of the architecture was designed, using anonymous subclasses is a _very_ Java/Kotlin platform idiomatic thing to do that doesn't translate well to the other languages. On Swift for example a single function/lambda callback with a boolean success parameter and then an optional error parameter is idiomatic and standard. If something similar to that is standard on TS/JS, we should do that instead of defining a `MParticleCallback` class. Instead we can typedef `MParticleCallback` to the callback function signature, which again would be idiomatic on iOS at least and I think also on Web.

Reply:
@einsteinx2 and I have decided to remove `MParticleCallback` and the related interfaces as the kotlin implementation is not idiomatic to Web or iOS. Also, as stated above, async/await should be the solution going forward, and Web should support promises natively or via polyfill.

Reply:
agreed